packages/core/src/shared/utilities/workspaceUtils.ts

```diff
@@ -315,6 +345,14 @@ export async function collectFiles(
         zipFilePath: string
     }[]
 > {
+    const workspaceSize = await getWorkspaceSize(sourcePaths, workspaceFolders, respectGitIgnore)
+    if (workspaceSize > maxSize) {
+        throw new ToolkitError(
+            'The project you have selected for source code is too large to use as context. Please select a different folder to use',
```

Comment:
This util is generic but this comment seems very specific to q

Reply:
If this can work elsewhere, we can explore abstracting it in the future

Reply:
It's already abstracted, that's why it lives in shared/util/.

The caller can pass an `maxSizeMsg` string.